pkg/loop/service.go

```diff
@@ -212,11 +212,43 @@ func (s *pluginService[P, S]) closeClient() (err error) {
 	return
 }
 
-func (s *pluginService[P, S]) wait(ctx context.Context) error {
+func (s *PluginService[P, S]) Wait(ctx context.Context) error {
 	select {
 	case <-ctx.Done():
 		return context.Cause(ctx)
 	case <-s.serviceCh:
 		return nil
+	case <-s.stopCh:
+		return nil
+	}
+}
+
+// XXXTestHook returns a TestPluginService.
+// It must only be called once, and before Start.
+func (s *PluginService[P, S]) XXXTestHook() TestPluginService[P, S] {
```

Comment:
IMHO we should avoid this at all costs. Have all other options been exhausted?

Reply:
Yes, I tried moving this to a `_test` package but that won't get exported to other packages.

One alternative is to expose `testInterrupt` instead, but I think that runs into the same problem.